Kudu.Core/Deployment/Generator/OryxBuilder.cs

```diff
@@ -65,7 +80,13 @@ public override Task Build(DeploymentContext context)
                 }
             }
 
-            return Task.CompletedTask;
+            // Detect if package upload is necessary for server side build
+            if (artifactPath != null)
+            {
```

Comment:
Please correct me if I am wrong, but looks like `artifactPath` will never actually be `null`. So, this path will always be hit, whether it's consumption or dedicated. We probably only want to hit this condition if we are in Consumption I think. Should we do a check for `FunctionAppHelper.HasScmRunFromPackage()` here?